File: movies-master/src/main/java/be/vdab/movies/klanten/Klant.java
package be.vdab.movies.klanten;

 class Klant {
    private final long id;
    private final String familienaam;
    private final String voornaam;
    private final String straatNummer;
    private final String postcode;
    private final String gemeente;

     Klant(long id, String familienaam, String voornaam, String straatNummer, String postcode, String gemeente) {
        this.id = id;
        this.familienaam = familienaam;
        this.voornaam = voornaam;
        this.straatNummer = straatNummer;
        this.postcode = postcode;
        this.gemeente = gemeente;
    }

     public long getId() {
        return id;
    }

     public String getFamilienaam() {
        return familienaam;
    }

     public String getVoornaam() {
        return voornaam;
    }

     public String getStraatNummer() {
        return straatNummer;
    }

    public String getPostcode() {
        return postcode;
    }

    public String getGemeente() {
        return gemeente;
    }
}

File: movies-master/src/main/resources/static/js/klant.js
"use strict";
import {byId, toon, verberg} from "./util.js";

byId("zoek").onclick = async () => {
    verberg("woordFout", "storing", "conflict", "badRequest", "mededeling");
    const woordInput = byId("woord");
    if (!woordInput.checkValidity()) {
        toon("woordFout");
    }
    const aantalFouten = document.querySelectorAll(".fout:not([hidden])").length;
    if (aantalFouten === 0) {
        zoek(woordInput.value);
    }
}

async function zoek(woord) {
    const klantResponse = await fetch(`klanten?woord=${woord}`);
    if (klantResponse.ok) {
        const klantenBody = byId("klantenBody");
        klantenBody.innerHTML = "";
        const klanten = await klantResponse.json();
        if (klanten.length > 0) {
            verberg("mededeling");
            toon("tabelHoofd");
            for (const klant of klanten) {
                const tr = klantenBody.insertRow();
                const td = tr.insertCell();
                const hyperlink = document.createElement("a");
                hyperlink.href = "bevestigpagina.html"
                hyperlink.textContent = klant.familienaam + " " + klant.voornaam;
                hyperlink.onclick = () => {
                    const klantInformatie = {
                        id: klant.id,
                        voornaam: klant.voornaam,
                        familienaam: klant.familienaam
                    }
                    sessionStorage.setItem("klant", JSON.stringify(klantInformatie));
                }
                td.appendChild(hyperlink);
                tr.insertCell().textContent = klant.straatNummer;
                tr.insertCell().textContent = klant.postcode;
                tr.insertCell().textContent = klant.gemeente;
            }
        } else {
            verberg("tabelHoofd");
            toon("mededeling")
        }
    }
}
